components/gateway/internal/auditlog/worker.go

```diff
@@ -29,12 +29,15 @@ func (w *Worker) Start() {
 	for {
 		select {
 		case <-w.done:
+			log.Println("Worker for auditlog message processing has finished")
 			return
 		case msg := <-w.auditlogChannel:
 			err := w.svc.Log(msg.Request, msg.Response, msg.Claims)
 			if err != nil {
-				log.Printf("error while saving auditlog: %s\n", err.Error())
+				log.Printf("error while saving auditlog message %+v with error: %s", msg, err.Error())
+				return
```

Comment:
this should be continue, otherwise you will stop the worker if just one message fails.